sdk/core/System.ClientModel/src/Internal/ClientUtilities.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ClientModel.Internal;
+
+internal class ClientUtilities
```

Comment:
There is a lot of duplication cause buy these argument validation helpers. Can we talk to the BCL team about solving this in the BCL?

Reply:
I agree with this, and it's true of the StreamExtensions class I just added to support the response buffering as well.  I think the question here is what timeline we want to address this on - i.e. do we want to focus on public APIs first and come back to internal types later?  Although I know it's hard to find time to pay down technical debt too.  Let's discuss the tradeoffs here.

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/41035